registry.go

```diff
@@ -149,21 +150,30 @@ func (r *Registry) Flush() error {
 		tags = append(tags, rtags...)
 
 		switch inst := val.(type) {
+
 		case Discrete:
 			val := inst.Snapshot()
+			if math.IsNaN(val) || math.IsInf(val, 0) {
+				break
+			}
 			for _, rep := range reporters {
 				if err := rep.Discrete(name, tags, val); err != nil {
 					return err
 				}
 			}
+
 		case Sample:
 			val := inst.Snapshot()
+			if val.Count() == 0 {
```

Comment:
That's not right. It means we will never report 0s

Reply:
Hm, @dim I thought that 0 count distribution means "nothing happened for metric X" so it shouldn't be reported.

What about this approach then? https://github.com/bsm/instruments/pull/9

Reply:
I think you should just remove this condition again and we are done!